internal/resources/metal/vlan/resource.go

```diff
@@ -81,96 +82,103 @@ func resourceMetalVlan() *schema.Resource {
 	}
 }
 
-func resourceMetalVlanCreate(d *schema.ResourceData, meta interface{}) error {
-	meta.(*config.Config).AddModuleToMetalUserAgent(d)
-	client := meta.(*config.Config).Metal
+func resourceMetalVlanCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	meta.(*config.Config).AddModuleToMetalGoUserAgent(d)
+	client := meta.(*config.Config).Metalgo
```

Comment:
I always trip myself up on updating these two lines.  If `config.Config` had `GetMetalClient(d *schema.ResourceData)`, `GetFabricClient(d *schema.ResourceData)`, etc. functions we could handle the User-Agent modification there so it doesn't have to be replicated in every CRUD function in every resource & data source.

Reply:
> every resource & data source

Turns out we're only modifying the User-Agent in data sources.  Why is that?